File: main-api/internal/utils/request.go
package utils

type CheckLimitAndOffsetParams struct {
	Limit        *uint32
	Offset       *uint32
	DefaultLimit uint32
}

func CheckLimitAndOffset(arg CheckLimitAndOffsetParams) (limit uint32, offset uint32) {
	var defaultLimit uint32 = arg.DefaultLimit

	if defaultLimit == 0 {
		defaultLimit = 20
	}

	o := uint32(0)
	l := defaultLimit

	if arg.Offset != nil && *arg.Offset > 0 {
		o = *arg.Offset
	}
	if arg.Limit != nil && *arg.Limit > 0 && *arg.Limit < defaultLimit {
		l = *arg.Limit
	}

	return l, o
}

File: api-gateway/src/utils/request.ts
export function checkLimitAndOffset(limit: any, offset: any): [number, number] {
    let l = Number(limit)
    let o = Number(offset)

    if (isNaN(l)) {
        l = 20
    }

    if (isNaN(o)) {
        o = 0
    }

    return [l, o]
}